src/Filter/ORM/TypeCaster.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+/**
+ * @see       https://github.com/laminas-api-tools/api-tools-doctrine-querybuilder for the canonical source repository
+ * @copyright https://github.com/laminas-api-tools/api-tools-doctrine-querybuilder/blob/master/COPYRIGHT.md
+ * @license   https://github.com/laminas-api-tools/api-tools-doctrine-querybuilder/blob/master/LICENSE.md New BSD License
+ */
+
+namespace Laminas\ApiTools\Doctrine\QueryBuilder\Filter\ORM;
+
+use DateTime;
+use Laminas\ApiTools\Doctrine\QueryBuilder\Filter\TypeCastInterface;
+
+class TypeCaster implements TypeCastInterface
+{
+    public function typeCastField($metadata, $field, $value, $format = null, $doNotTypecastDatetime = false)
+    {
+        if (! isset($metadata->fieldMappings[$field])) {
+            return $value;
+        }
+
+        switch ($metadata->fieldMappings[$field]['type']) {
+            case 'string':
+                settype($value, 'string');
+                break;
+            case 'integer':
```

Comment:
Do we need `int` here as well?

Reply:
Nope, `int` is not doctrine mapping type.